index.bs

```diff
@@ -439,29 +449,33 @@ interface FontManager {
 1. If [=/this=]’s [=relevant global object=]'s [=/associated Document=] is not [=/allowed to use=] the [=/policy-controlled feature=] named {{PermissionPolicy/"local-fonts"}}, then [=/reject=] |promise| with a "{{SecurityError}}" {{DOMException}}, and return |promise|.
 1. If [=/this=]’s [=relevant global object=] does not have [=/transient activation=], then [=/reject=] |promise| with a "{{SecurityError}}" {{DOMException}}, and return |promise|.
 1. Otherwise, run these steps [=in parallel=]:
-    1. Let |permissionState| be [=request permission to use=] {{PermissionName/"font-access"}}.
-    1. If |permissionState| is {{PermissionState/"denied"}}, then [=/reject=] |promise| with a "{{NotAllowedError}}" {{DOMException}}, and abort these steps.
-    1. Let |fonts| be a [=/list=] of all local fonts on the system.
-    1. Let |select| be |options|' {{QueryOptions/"select"}} member.
-    1. Let |result| be an new [=/list=].
-    1. [=list/For each=] font |font| in |fonts|, run these steps:
+    1. Let |system fonts| be a [=/list=] of all local fonts on the system.
+    1. Let |selectable fonts| be a new [=/list=].
+    1. [=list/For each=] font |font| in |system fonts|, run these steps:
         1. Let |representation| be a [=/font representation=] for |font|.
-        1. Let |postscriptName| be |representation|'s [=font representation/name string=] 6 for \``en`\`.
-        1. If |select| is empty, or if |select| [=list/contains=] |postscriptName|, then [=list/append=] a new {{FontMetadata}} instance associated with |representation| to |result|.
-    1. Sort |list| in [=list/sort in ascending order|ascending order=] by using {{FontMetadata/postscriptName}} as the sort key and store the result as |list|.
-    1. [=/Resolve=] |promise| with |list|.
+        1. Set |representation|'s [=font representation/data bytes=] to a SFNT [[!SFNT]] serialization of the representation.
+        1. Let |postscriptName| be |representation|'s [=font representation/PostScript name=].
+        1. If |postscriptName| is not valid according to the [[!OPENTYPE]] <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/name">naming table</a> specification, then [=iteration/continue=].
+        1. If |options|[{{QueryOptions/"postscriptNames"}}] [=map/exists=] and |options|[{{QueryOptions/"postscriptNames"}}] does not [=list/contain=] |postscriptName|, then [=iteration/continue=].
+        1. [=list/Append=] a new {{FontMetadata}} instance associated with |representation| to |selectable fonts|.
+    1. [=/Prompt the user to choose=] one or more items from |selectable fonts|, with |descriptor|, and let |result| be the result.
+        <aside class=note>User agents that choose to make the permission a binary choice rather than displaying a picker can have the permission call return all local fonts on the system.</aside>
```

Comment:
```suggestion
        User agents may present a yes/no choice instead of a list of choices, and in that case they should set |result| to |selectable fonts|.
```

Note that "all local fonts" wasn't right given the filtering above.

I also made this normative, because https://w3c.github.io/permissions/#prompt-the-user-to-choose is reasonably clear about the UA showing a list of choices, so we need to override that. I wrote "should" instead of "must" because UAs might want to subset that list, although the above question about UAs eliminating fonts might cover this.